dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyClient.java

```diff
@@ -89,23 +99,23 @@ public NettyClient(final URL url, final ChannelHandler handler) throws RemotingE
     protected void doOpen() throws Throwable {
         final NettyClientHandler nettyClientHandler = new NettyClientHandler(getUrl(), this);
         bootstrap = new Bootstrap();
-        bootstrap.group(nioEventLoopGroup)
+        bootstrap.group(eventLoopGroup)
```

Comment:
if we have two factory methods, then we can change this logic to 

```java
        bootstrap.group(eventLoopGroup())
                .option(ChannelOption.SO_KEEPALIVE, true)
                .option(ChannelOption.TCP_NODELAY, true)
                .option(ChannelOption.ALLOCATOR, PooledByteBufAllocator.DEFAULT)
                //.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, getTimeout())
                .channel(socketChannelClass());
```

socketChannelClass()

```java
    private static Class<? extends SocketChannel> socketChannelClass() {
        String osName = SystemPropertyUtil.get("os.name").toLowerCase(Locale.ENGLISH).trim();
        return "linux".equals(osName) ? EpollSocketChannel.class : NioSocketChannel.class)
    }
```